File: Sorting/insettionSort/insertionSort.cpp
#include <iostream>
using namespace std;
void insertionSort(int arr[],int n){
    for(int i = 0;i<n; i++){
        int curr = arr[i];
        int j = i-1;
        while(j>=0 && curr < arr[j]){
            arr[j+1] = arr[j];
            j--;
    }
    arr[j+1] = curr;
}
}

void printArr(int arr[],int n){
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
}
int main(){
    int arr[] = {5,4,3,8,6,9};
    int n = sizeof(arr)/sizeof(arr[0]);
    insertionSort(arr,n);
    printArr(arr,n);

    return 0;
}

File: Sorting/insettionSort/InsertionSort.java
class InsertionSort{
    public static void insertionSort(int arr[]){
        for(int i=0; i<arr.length;i++){
            int curr = arr[i];
            int j = i-1;
            while(j>=0 && curr < arr[j]){
                arr[j+1] = arr[j];
                j--;
            }
            arr[j+1] = curr;
        }
    }

    public static void printArr(int arr[]){
        for(int i = 0; i < arr.length; i++){
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
    public static void main(String[] args) {
        int arr[] = {3,2,5,6,8,4};

        insertionSort(arr);
        printArr(arr);

    }
}